### DISCUSSION TITLE
Tables and columns naming conventions

### DISCUSSION THREAD

**Discussion by Vadorequest** [supabase/supabase#619] [Questions] [answered]
Intuitively I started naming my tables using `PascalCasing` and columns using `camelCasing`.

![image](https://user-images.githubusercontent.com/3807458/106028669-853e9c80-60cc-11eb-82f4-c1e6e784039c.png)

I see all examples use `snake_casing`, is that a Postgre convention? Should I follow it? Does it changes anything, or is it just a personal preference?

**Comment by steve-chavez** [comment, marked as answer]
It's a PostgreSQL convention. The main drawback of using camel case is that you'll need to double quote your tables when using SQL:

```sql
SELECT "myCol1", "myCol2", "myCol3" FROM "myTable1";

-- with snake case you don't need to double quote
-- SELECT my_col_1, my_col_2, my_col_3 FROM my_table_1;
```

Which results in more typing, specially if you plan on doing complex SQL.

However it's fine to use camel case if you prefer so, all our components should handle different types of identifiers(even with unicode) just fine.

**Reply by wiesson** [reply]
Thanks!

Is there an option to auto-map from `snake_case` to `camelCase` in Postgrest? I would be great if the database could follow their convention and the frontend as well

**Reply by steve-chavez** [reply]
@wiesson See https://github.com/supabase/postgrest-js/issues/204

**Comment by esDotDev** [comment]
Ah, ran into this last night and thought I had found a bug. _thats_ why I needed the quotes. Hmm...
